pyquil/quilatom.py

```diff
@@ -155,7 +163,23 @@ def register(cls, n: int) -> List["QubitPlaceholder"]:
         return [cls() for _ in range(n)]
 
 
-QubitDesignator = Union[Qubit, QubitPlaceholder, FormalArgument, int]
+QubitDesignator = Union[Qubit, QubitPlaceholder, FormalArgument, RSQubit, int]
+
+
+def convert_to_rs_qubit(qubit: QubitDesignator) -> RSQubit:
+    if isinstance(qubit, Qubit):
+        return RSQubit.from_fixed(qubit.index)
+    if isinstance(qubit, QubitPlaceholder):
+        raise NotImplementedError("QubitPlaceholders aren't implemented in quil-rs")
+    if isinstance(qubit, FormalArgument):
+        return RSQubit.from_variable(qubit.name)
+    if isinstance(qubit, int):
+        return RSQubit.from_fixed(qubit)
+    return qubit
```

Comment:
These should have an `else` that raises because the type hint is not enforced

Reply:
Fixed